### ISSUE TITLE
Inconsistencies in `action_dim`, `action_horizon`, and `max_token_len` between `pi0` and `pi0_fast` Configurations on the LIBERO Dataset

### ISSUE THREAD

**Issue by zheruiqiu** [Physical-Intelligence/openpi#350]
Hello,

I've noticed some inconsistencies in the configuration settings between the `pi0` and `pi0_fast` models, specifically regarding the `action_dim`, `action_horizon`, and `max_token_len` parameters on the LIBERO dataset. Here are the details:

| **Parameter**     | $\pi_0$ | $\pi_0$-FAST |
|-------------------|---------|--------------|
| **`action_dim`**  | 32      | 7            |
| **`action_horizon`** | 50      | 10           |
| **`max_token_len`** | 48      | 180          |

As referenced in:
 - for $\pi_0$
https://github.com/Physical-Intelligence/openpi/blob/d0b6231bd32f1bc6a860b7366e991cfae11a6a84/src/openpi/training/config.py#L485
https://github.com/Physical-Intelligence/openpi/blob/d0b6231bd32f1bc6a860b7366e991cfae11a6a84/src/openpi/models/pi0.py#L74-L76
 - for $\pi_0$ - FAST
https://github.com/Physical-Intelligence/openpi/blob/d0b6231bd32f1bc6a860b7366e991cfae11a6a84/src/openpi/training/config.py#L541

These differences seem significant, and I'm curious about the rationale behind having different configurations for these parameters. Is there a specific reason for not unifying these settings across both models? Understanding the design choices here would be very helpful.

Additionally, given that there are discrepancies in the configuration settings, was the model comparison in the paper conducted under these different configurations? It would be helpful to know if the comparison results reflect these inconsistencies in the settings.

Thank you for your time and assistance!

**Comment by kpertsch** [comment]
Hi, 

To understand the difference, we need to understand the different architectures of these models. pi0 is a diffusion VLA that has a fixed output dimension (action dim + action horizon), so we can't change that from pre-training. So we use the same action horizon as during pre-training (50), and for the action dim we pad to the dimension used during pre-training (32).

On the other hand, pi0-FAST is an autoregressive model that can flexibly decode outputs of different sizes (ie token sequence lengths). So for pi0-FAST we do not pad the action dimension and use a shorter chunk length that's more in line with the control frequency of the robot. We have not ablated whether using the longer action chunk would improve performance -- generally, our goal here was not to compare the two model architectures, but to provide a simple example how to finetune each model -- we have not done any hyperparameter tuning.

As for the max_token_len: for pi0 the tokens only contain the prompt, for pi-FAST they also contain the target action tokens, so naturally the max_sequence_len needs to be higher for the latter. This does not affect training dynamics, ie setting the max token len for pi0 to 180 would just make memory consumption higher, but not change the outcome.

Hope this helps to answer your question!

**Comment by zheruiqiu** [comment]
Thank you so much for the detailed and thoughtful response! I really appreciate you taking the time to clarify this.

Thanks again for your help!

**Comment by cgyrobot** [comment]
thanks，I meet this too
